client/src/pages/GalleryPage.js

```diff
@@ -58,25 +56,31 @@ const GalleryPage = () => {
               <Button
                 label="◀"
                 type="submit"
-                onClick={previousImage}
-                disabled={indexOfCurrentImage === 0}
+                onClick={loadPreviousImage}
+                disabled={!previousImage}
               />
-              {loading && <p>Loading...</p>}
-              {error && <p>{error.message}</p>}
-              {currentImage && <img src={currentImage.url} alt="" />}
+              {isLoading && <p>Loading...</p>}
+              {isError && <p>{error}</p>}
+              {selectedImage && <img src={selectedImage.url} alt="" />}
               <Button
                 label="▶"
                 type="submit"
-                onClick={nextImage}
-                disabled={indexOfCurrentImage === userData.images.length - 1}
+                onClick={loadNextImage}
+                disabled={!nextImage}
               />
             </ImageContainer>
           </ImageDisplay>
           <TagDisplay>
             <p>Tag-Display</p>
-            {currentImage &&
-              currentImage.tags.map((tag, index) => (
-                <Button key={index} label={tag} />
+            {selectedImage &&
+              selectedImage.tags.map((tag) => (
+                <Button
+                  key={tag}
```

Comment:
"key" should be unique, so if you have something like an "image-id" I would recommend using that here instead of tag

Reply:
This is complicated; originally I wanted to use `index` als key, but was recommended not to do so: it might blow up things when the the array gets filtered. My tags don't have any ids. As one tag should be assigned only once, this is a provisional solution.